### ISSUE TITLE
Can this generate splash-screen

### ISSUE THREAD

**Issue by Barba828** [vite-pwa/assets-generator#7]
https://web.dev/learn/pwa/enhancements/#splash-screens
Otherwise I may need to use other tools to generate these static resources

**Comment by userquin** [comment]
Not yet, we need to start the app on dev mode, maybe we can add some tool/plugin to generate them.

**Comment by tripflex** [comment]
yeah this would be an amazing addition to have

**Comment by userquin** [comment]
@tripflex @Barba828 I'm working on this, I'll try to send a draft PR tmr, but I'll need your help, rn I have not any iOS/macOS device to test the splash screens

**Comment by userquin** [comment]
![imagen](https://github.com/vite-pwa/assets-generator/assets/6311119/3546235e-a442-4655-b4ec-3e51df2d46be)

**Comment by userquin** [comment]
@Barba828 released v0.0.5 version, can you test it in your project?

**Comment by userquin** [comment]
@Barba828 I'll need to review a few things in sharp usage about `density` in the options (at least we've an initial version)

**Comment by userquin** [comment]
@Barba828 can you check this https://deploy-preview-74--vite-pwa-org.netlify.app/assets-generator/cli.html#ios-ipad-splash-screens ? the PR can be found here https://github.com/vite-pwa/docs/pull/74

**Comment by Barba828** [comment]
I've upgraded to version 0.0.5, but I'm not generating a slash screen after changing the `pwa-assets.config.ts`!

**Comment by userquin** [comment]
@Barba828 it is about you dont' include the entry or about the CLI not generating the splash screens

**Comment by Barba828** [comment]
@userquin It looks as if there is a problem with the judgment [here](https://github.com/vite-pwa/assets-generator/blob/538213780aa6177390089a91480a7100eec493e7/src/cli-start.ts#L76C7-L76C16), here the preset I got from config doesn't have `appleSplashScreens` 
<img width="703" alt="image" src="https://github.com/vite-pwa/assets-generator/assets/29791171/b8ada223-8b17-4575-9a89-aeb39dd8fda3">

**Comment by userquin** [comment]
@Barba828 ok, checking what's happening...

**Comment by userquin** [comment]
@Barba828 is your repo here https://github.com/Barba828/buitar/blob/main/packages/buitar/pwa-assets.config.ts ?

**Comment by Barba828** [comment]
Yes，And I've upgraded to version 0.0.5 locally

**Comment by userquin** [comment]
@Barba828 can you provide the configuration for the splash screens? Or just push the changes to the repo in another branch/PR.

**Comment by userquin** [comment]
![imagen](https://github.com/vite-pwa/assets-generator/assets/6311119/36338894-f53c-44b5-b838-713c343b8b27)

**Comment by Barba828** [comment]
I'm using the default configuration from docs. And I am able to generate images by getting the appleSplashScreens directly from config
```
  const {
    appleSplashScreens: useAppleSplashScreens
  } = config;
```

**Comment by userquin** [comment]
Forked your repro adding apple splash screens entry once updated the dev dependency:

![imagen](https://github.com/vite-pwa/assets-generator/assets/6311119/649f7ef7-b12a-4afa-9192-40701be01c43)

**Comment by userquin** [comment]
@Barba828 have you run `pnpm install` from root once dev dependency update?

**Comment by Barba828** [comment]
I've found what's wrong with me.
```
export default defineConfig({
	preset: minimalPreset,        // I need to configure it in this object
	images: ['public/icons/logo.png'],
        
        // Configuring it here will not generate resources
	appleSplashScreens: createAppleSplashScreens({
		padding: 0,
	}),
})
```

**Comment by userquin** [comment]
Anyway, I need to review why not generating all the png files, maybe memory problems, rn only generating 15 (we've 39 devices/sizes)

I guess it is about devices entries using same sizes.

**Comment by userquin** [comment]
Confirmed, check for example these devices:
- `'iPad Pro 11"': { width: 1668, height: 2388, scaleFactor: 2 },` and `'iPad Pro 10.5"': { width: 1668, height: 2388, scaleFactor: 2 },`
- `'iPad Pro 9.7"': { width: 1536, height: 2048, scaleFactor: 2 },`, `'iPad mini 7.9"': { width: 1536, height: 2048, scaleFactor: 2 },`, `'iPad Air 9.7"': { width: 1536, height: 2048, scaleFactor: 2 },` and `'iPad 9.7"': { width: 1536, height: 2048, scaleFactor: 2 },`
- `'iPhone 14 Plus': { width: 1284, height: 2778, scaleFactor: 3 },`, `'iPhone 13 Pro Max': { width: 1284, height: 2778, scaleFactor: 3 },` and `'iPhone 12 Pro Max': { width: 1284, height: 2778, scaleFactor: 3 },`

**Comment by userquin** [comment]
@Barba828 I'll filter duplicated entries in PNG generation and also in links, in the meantime can you test the splash screen in the iOS emulator or real device?

**Comment by userquin** [comment]
@Barba828 this no makes sense, 15KB only in links to each html page:

![imagen](https://github.com/vite-pwa/assets-generator/assets/6311119/28425cb8-373a-4f84-9e04-4f525c8e5a4a)

**Comment by Barba828** [comment]
OK, It looks like splash-screen is working fine on real device.

**Comment by userquin** [comment]
I'll release a new patch version (0.0.6) filtering the device dimensions, sending PR...

**Comment by userquin** [comment]
> OK, It looks like splash-screen is working fine on real device.

Do you have the url? I'll send it to some friend and maybe we can test in iOS emulator...

**Comment by userquin** [comment]
@Barba828 released new `0.0.6` version.

**Comment by userquin** [comment]
@Barba828 I also need to update the docs, it is wrong the way we register the Apple splash screens object, it should be moved to the preset.

**Comment by userquin** [comment]
I add a new helper function `combinePresetAndAppleSplashScreens` (released new 0.0.7 version), the docs also updated to fix the `appleSplashScreens` usage and including the new `combinePresetAndAppleSplashScreens`.

That's all for today, see you

**Comment by Barba828** [comment]
By the way, I have a question. Why only log `link` in [here](https://github.com/vite-pwa/assets-generator/blob/878493de0e27a2f150d7dd0a6e99e64bd94b8a8e/src/build.ts#L67) . Can it be written directly into the index.html file after I set entry(or other or other fieldname).
```ts
export interface AppleTouchStartupImageOptions {
  /**
   * Write <link> into html 
   */
  entry?: string
}
```
